### PR TITLE
Updated the Infoblox member module

### PR DESCRIPTION
**Pull request by krisvasudevan**
##### SUMMARY

Updated the nios_member.py and api.py to support the function create_token

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

nios_member

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/module_utils/net_tools/nios/api.py [hunk 1]
```diff
@@ -442,6 +445,14 @@ def get_object_ref(self, module, ib_obj_type, obj_filter, ib_spec):
             # reinstate restart_if_needed key if it's set to true in play
             if module.params['restart_if_needed']:
                 ib_spec['restart_if_needed'] = temp
+        elif (ib_obj_type == NIOS_MEMBER):
+            # del key 'create_token' as nios_member get_object fails with the key present
+            temp = ib_spec['create_token']
+            del ib_spec['create_token']
+            ib_obj = self.get_object(ib_obj_type, obj_filter.copy(), return_fields=ib_spec.keys())
```
Comment (justjais) [inline lib/ansible/module_utils/net_tools/nios/api.py:453]: `get_object` functionality should be handled under `get_object_ref`.

File: lib/ansible/module_utils/net_tools/nios/api.py [hunk 2]
```diff
@@ -25,7 +25,6 @@
 # LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 # USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
-
```
Comment (justjais) [inline lib/ansible/module_utils/net_tools/nios/api.py]: please edit these changes

**Review by justjais — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/net_tools/nios/api.py (hunk 1 above)
Comment (krisvasudevan) [reply lib/ansible/module_utils/net_tools/nios/api.py:453]: I am not clear about this comment

Comment (justjais) [reply lib/ansible/module_utils/net_tools/nios/api.py:453]: Check `get_object_ref` function in `api.py` where `self.get_object` functionality is handled, so please move the resepctive functionality under `get_object_ref`.

Comment (krisvasudevan) [reply lib/ansible/module_utils/net_tools/nios/api.py:453]: I have added this functionality under the get_object_ref function
`
    def get_object_ref(self, module, ib_obj_type, obj_filter, ib_spec):
        ''' this function gets the reference object of pre-existing nios objects '''

        update = False
        old_name = new_name = None
        if ('name' in obj_filter):
            # gets and returns the current object based on name/old_name passed
            try:
                name_obj = self.module._check_type_dict(obj_filter['name'])
                old_name = name_obj['old_name']
                new_name = name_obj['new_name']
            except TypeError:
                name = obj_filter['name']

            if old_name and new_name:
                if (ib_obj_type == NIOS_HOST_RECORD):
                    test_obj_filter = dict([('name', old_name), ('view', obj_filter['view'])])
                elif (ib_obj_type in (NIOS_AAAA_RECORD, NIOS_A_RECORD)):
                    test_obj_filter = obj_filter
                else:
                    test_obj_filter = dict([('name', old_name)])
                # get the object reference
                ib_obj = self.get_object(ib_obj_type, test_obj_filter, return_fields=ib_spec.keys())
                if ib_obj:
                    obj_filter['name'] = new_name
                else:
                    test_obj_filter['name'] = new_name
                    ib_obj = self.get_object(ib_obj_type, test_obj_filter, return_fields=ib_spec.keys())
                update = True
                return ib_obj, update, new_name
            if (ib_obj_type == NIOS_HOST_RECORD):
                # to check only by name if dns bypassing is set
                if not obj_filter['configure_for_dns']:
                    test_obj_filter = dict([('name', name)])
                else:
                    test_obj_filter = dict([('name', name), ('view', obj_filter['view'])])
            elif (ib_obj_type == NIOS_IPV4_FIXED_ADDRESS or ib_obj_type == NIOS_IPV6_FIXED_ADDRESS and 'mac' in obj_filter):
                test_obj_filter = dict([['mac', obj_filter['mac']]])
            elif (ib_obj_type == NIOS_A_RECORD):
                # resolves issue where a_record with uppercase name was returning null and was failing
                test_obj_filter = obj_filter
                test_obj_filter['name'] = test_obj_filter['name'].lower()
            # check if test_obj_filter is empty copy passed obj_filter
            else:
                test_obj_filter = obj_filter
            ib_obj = self.get_object(ib_obj_type, test_obj_filter.copy(), return_fields=ib_spec.keys())
        elif (ib_obj_type == NIOS_ZONE):
            # del key 'restart_if_needed' as nios_zone get_object fails with the key present
            temp = ib_spec['restart_if_needed']
            del ib_spec['restart_if_needed']
            ib_obj = self.get_object(ib_obj_type, obj_filter.copy(), return_fields=ib_spec.keys())
            # reinstate restart_if_needed key if it's set to true in play
            if module.params['restart_if_needed']:
                ib_spec['restart_if_needed'] = temp
        elif (ib_obj_type == NIOS_MEMBER):
            # del key 'create_token' as nios_member get_object fails with the key present
            temp = ib_spec['create_token']
            del ib_spec['create_token']
            ib_obj = self.get_object(ib_obj_type, obj_filter.copy(), return_fields=ib_spec.keys())
            if temp:
                # reinstate 'create_token' key
                ib_spec['create_token'] = temp`

Comment (justjais) [reply lib/ansible/module_utils/net_tools/nios/api.py:453]: ah.. ignore the comment then.

**Review by justjais — APPROVED** [review]

**Review by justjais — APPROVED** [review]

**Pull request merged**